resteasy/src/test/scala/com/cerner/beadledom/resteasy/ModuleProcessorSpec.scala

```diff
@@ -0,0 +1,84 @@
+package com.cerner.beadledom.resteasy
+
+import com.cerner.beadledom.resteasy.fauxservice.FauxModule
+import org.jboss.resteasy.spi.{Registry, ResteasyProviderFactory}
+import org.scalatest.{FunSpec, MustMatchers}
+import org.scalatest.mockito.MockitoSugar
+import com.google.inject.{AbstractModule, Guice}
+
+class ModuleProcessorSpec extends FunSpec with MockitoSugar with MustMatchers{
+
+  val registry: Registry = mock[Registry]
+  val providerFactory: ResteasyProviderFactory = mock[ResteasyProviderFactory]
+
+  describe("ModuleProcessor") {
+
+    describe("constructor") {
+      it("throws a NullPointerException for null registry") {
+        intercept[NullPointerException] {
+          new ModuleProcessor(null, providerFactory)
+        }
+      }
+
+      it("throws a NullPointerException for null provider factory") {
+        intercept[NullPointerException] {
+          new ModuleProcessor(registry, null)
+        }
+      }
+    }
+
+    describe("ModuleProcessor#processInjector") {
+      it("throws a NullPointerException for null injector") {
+        intercept[NullPointerException] {
+          new ModuleProcessor(registry, providerFactory).processInjector(null)
+        }
+      }
+
+      it("excludes providers that have a binding annotation") {
+        val injector = Guice.createInjector(new AbstractModule {
+          override def configure(): Unit = {
+            bind(classOf[TestJaxRsProvider]).annotatedWith(classOf[TestBindingAnnotation]).toProvider(
+              new TestProviderWithBindingAnnotation(classOf[TestBindingAnnotation])
+            )
+          }
+        })
+        val providerFactory = new ResteasyProviderFactory()
+        new ModuleProcessor(registry, providerFactory).processInjector(injector)
+
+        providerFactory.getProviderInstances mustBe empty
+      }
+
+      it("excludes provider with binding annotation when two providers of the same type are bound with the injector") {
+        val injector = Guice.createInjector(new AbstractModule {
+          override def configure(): Unit = {
+            bind(classOf[TestJaxRsProvider]).annotatedWith(classOf[TestBindingAnnotation]).toProvider(
```

Comment:
The call to `toProvider` here and elsewhere aren't needed. Guice can create an instance of the binding via a constructor, and if there are multiple constructors it looks for the `@Inject` annotation to know which one to use.

Reply:
https://github.com/cerner/beadledom/pull/143#discussion_r301580313 Are you suggesting i have to change the code in the TestJaxRxProvider class in some way?

Reply:
No. I took a look at the Guice annotated binding docs [1] and it looks like you can't do it without some `#to` call. You should be able to do `.to(classOf[TestJaxRsProvider])` rather than having a custom Guice provider though.

[1] https://github.com/google/guice/wiki/BindingAnnotations

Reply:
https://github.com/cerner/beadledom/pull/143/commits/baf1e83e23496f193739668eb9632289d1cfbc1c